### PR TITLE
[Merged by Bors] - chore: remove obsolete porting notes

### PR DESCRIPTION
**Pull request by adomani**
I should have removed these porting notes in

* #13982,

but the replacement had been done automatically.

Luckily, Ruben noticed and flagged this while/after the PR was merging, but at a time when I did not have the capacity to revisit!

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Category/UniformSpace.lean [hunk 1]
```diff
@@ -65,15 +65,11 @@ theorem coe_of (X : Type u) [UniformSpace X] : (of X : Type u) = X :=
 instance (X Y : UniformSpaceCat) : CoeFun (X ⟶ Y) fun _ => X → Y :=
   ⟨(forget UniformSpaceCat).map⟩
 
--- Porting note: `simpNF` should not trigger on `rfl` lemmas.
--- see https://github.com/leanprover/std4/issues/86
```
Comment (grunweg) [inline Mathlib/Topology/Category/UniformSpace.lean:69]: So can/should this issue be closed?

File: Mathlib/NumberTheory/ModularForms/CongruenceSubgroups.lean [hunk 2]
```diff
@@ -148,7 +148,7 @@ def Gamma1' (N : ℕ) : Subgroup (Gamma0 N) :=
   (Gamma0Map N).ker
 #align Gamma1' Gamma1'
 
-@[simp] -- Porting note: linter failed to synth `CommMonoid { x // x ∈ Gamma0 N }`
+@[simp]
```
Comment (grunweg) [inline Mathlib/NumberTheory/ModularForms/CongruenceSubgroups.lean:151]: This error *looks* different (but presumably is fine to remove/perhaps this was more obvious with the original change).

File: Mathlib/NumberTheory/KummerDedekind.lean [hunk 3]
```diff
@@ -224,8 +224,6 @@ noncomputable def quotAdjoinEquivQuotMap (hx : (conductor R x).comap (algebraMap
     · exact Ideal.Quotient.mk_surjective
 #align quot_adjoin_equiv_quot_map quotAdjoinEquivQuotMap
 
--- Porting note: on-line linter fails with `failed to synthesize` instance
--- but #lint does not report any problem
```
Comment (grunweg) [inline Mathlib/NumberTheory/KummerDedekind.lean:228]: Let me also flag this one: this was nolint simpNF before, right? Out of context, this change looks slightly odd.

**Review by grunweg — COMMENTED** [review]
All of these are clearly stale now (I flagged three places where I trust your judgement, but the change in isolation looks less obvious). I thank you for following up on this!
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**